openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java

```diff
@@ -1102,4 +1106,37 @@ public static boolean isIntegerSchema(Schema<?> fieldSchema) {
     public static boolean isNumberSchema(Schema<?> fieldSchema) {
         return Objects.equals(GeneratorUtils.getOpenAPIType(fieldSchema), NUMBER);
     }
+
+    public static String resolveReferenceType(Schema<?> schema, Components components, boolean isWithoutDataBinding,
+                                              String pathParam) throws BallerinaOpenApiException {
+        String type = GeneratorUtils.extractReferenceType(schema.get$ref());
+
+        if (isWithoutDataBinding) {
+            Schema<?> referencedSchema = components.getSchemas().get(getValidName(type, true));
+            if (referencedSchema != null) {
+                if (referencedSchema.get$ref() != null) {
+                    type = resolveReferenceType(referencedSchema, components, isWithoutDataBinding, pathParam);
+                } else {
+                    type = getPathParameterType(referencedSchema, pathParam);
+                }
+            }
+        } else {
+            type = getValidName(type, true);
+        }
+        return type;
+    }
+
+    private static String getPathParameterType(Schema<?> typeSchema, String pathParam)
+            throws BallerinaOpenApiException {
+        String type;
+        if (!(isStringSchema(typeSchema) || isNumberSchema(typeSchema) || isBooleanSchema(typeSchema)
+                || isIntegerSchema(typeSchema))) {
+            type = "string";
```

Comment:
Don't we have a common constant for Ballerina string type?

Reply:
Updated with the constant value [cf982ff](https://github.com/ballerina-platform/openapi-tools/pull/1587/commits/cf982ffe6f3bfdbdce1dcd6b02ba6aae0acfc952)